### ISSUE TITLE
Want Style-guide

### ISSUE THREAD

**Issue by louis-langholtz** [louis-langholtz/PlayRho#2]
A source code style guide is needed.

Among other things, it should require formatting to be using 4-spaces instead of tabs. I've put `.editorconfig` [EditorConfig](http://editorconfig.org/) files into the source code trees now to help with this. Seems like a clang format file would be ideal for this.

**Comment by louis-langholtz** [comment]
I'm of the opinion that things which are automatically detectable make better elements of a style-guide. That's not to say that only automatically testable/checkable things should be in this guide; it's just to say that I'd prefer this guide to have objective/measurable guidelines.

**Comment by louis-langholtz** [comment]
Along the lines of having consistent style, I've recently added a [`.clang-format` file](https://github.com/louis-langholtz/PlayRho/blob/master/.clang-format) to the project/repo.

This was the first time I'd checked out clang format. I was surprised and disappointed to learn that clang-format doesn't have more configurability in this file. I tried to select settings that were closer to the current formatting conventions or desires I have for formatting. There were differences however.

I do think using clang-format is a step forward though from using EditorConfig. clang-format at least seems to be a complete superset of what EditorConfig appears able to be used for.
